lib/services/database.go

```diff
@@ -225,6 +228,58 @@ func tagsToLabels(tags []*rds.Tag) map[string]string {
 	return labels
 }
 
+// IsRDSClusterSupported checks whether the aurora cluster is supported and logs
+// related info if not.
+func IsRDSClusterSupported(cluster *rds.DBCluster, log logrus.FieldLogger) bool {
+	switch aws.StringValue(cluster.EngineMode) {
+	// Aurora Serverless (v1 and v2) does not support IAM authentication
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-2.limitations.html
+	case RDSEngineModeServerless:
+		log.Infof("Aurora cluster %q is %s which doesn't support IAM authentication. Skipping.",
+			aws.StringValue(cluster.DBClusterIdentifier),
+			aws.StringValue(cluster.EngineMode),
+		)
+		return false
+
+	// Aurora MySQL 1.22.2, 1.20.1, 1.19.6, and 5.6.10a only: Parallel query doesn't support AWS Identity and Access Management (IAM) database authentication.
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-mysql-parallel-query.html#aurora-mysql-parallel-query-limitations
+	case RDSEngineModeParallelQuery:
+		if apiutils.SliceContainsStr([]string{"1.22.2", "1.20.1", "1.19.6", "5.6.10a"}, auroraMySQLVersion(cluster)) {
```

Comment:
Is `parallelquery` mode only valid for MySQL engine?

Also, if we're checking the substring anyway, do we really need to try and parse our the aurora version?

Reply:
- yes, `parrallelquery` is only for Aurora MySQL
- i am currently comparing the full version string. i feel it's safer than substring.